pkg/rpc/server.go

```diff
@@ -0,0 +1,131 @@
+package smrpc
+
+import (
+	fmt "fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/google/fscrypt/filesystem"
+	"github.com/sirupsen/logrus"
+	"k8s.io/mount-utils"
+
+	empty "github.com/golang/protobuf/ptypes/empty"
+	"golang.org/x/net/context"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+
+	iscsiutil "github.com/longhorn/go-iscsi-helper/util"
+
+	"github.com/longhorn/longhorn-share-manager/pkg/server"
+	"github.com/longhorn/longhorn-share-manager/pkg/types"
+	"github.com/longhorn/longhorn-share-manager/pkg/util"
+	"github.com/longhorn/longhorn-share-manager/pkg/volume"
+)
+
+type ShareManagerServer struct {
+	manager *server.ShareManager
+}
+
+func NewShareManagerServer(manager *server.ShareManager) *ShareManagerServer {
+	return &ShareManagerServer{
+		manager: manager,
+	}
+}
+
+func (s *ShareManagerServer) FilesystemTrim(ctx context.Context, req *FilesystemTrimRequest) (resp *empty.Empty, err error) {
+	volumeName := s.manager.Volume.Name
+
+	defer func() {
+		if err != nil {
+			logrus.WithError(err).Errorf("failed to trim mounted filesystem on volume %v", volumeName)
+		}
+	}()
+
+	devicePath := types.GetVolumeDevicePath(volumeName, req.IsEncryptedDevice)
+	if !volume.CheckDeviceValid(devicePath) {
+		return &empty.Empty{}, fmt.Errorf("volume %v is not valid", volumeName)
```

Comment:
Can we use gRPC status with error instead? then we can return the error with a meaningful gRPC satus code.

https://pkg.go.dev/google.golang.org/grpc/status#Errorf

In this case, probably `FailedPrecondition` is meaningful. Do the same for the following returned errors.

Reply:
ref: https://github.com/longhorn/longhorn/issues/5150

cc @longhorn/dev

Reply:
Addressed. Thank you.